include/picongpu/fields/FieldManipulator.kernel

```diff
@@ -19,93 +19,199 @@
+            DataSpace< simDim > const localDomainCells = mapper.getGridSuperCells() * SuperCellSize::toRT();
+
+            using SuperCellDomCfg = IdxConfig<
+                cellsPerSuperCell,
+                numWorkers
+            >;
+
+            ForEachIdx<
+                SuperCellDomCfg
+            >{ workerIdx }(
+                [&](
+                    uint32_t const linearIdx,
+                    uint32_t const
+                )
+                {
+                    /* cell index within the superCell */
+                    DataSpace< simDim > const cellIdx = DataSpaceOperations< simDim >::
+                        template map< SuperCellSize >( linearIdx );
+
+                    DataSpace< simDim > cell( superCellIdx * SuperCellSize::toRT( ) + cellIdx);
+
+                    do
+                    {
+                        cell[ axis ] += SuperCellSize::toRT()[ axis ] * -1 * relExchangeDir[ axis ];
+                        int factor(0);
+
+                        if( relExchangeDir[ axis ] < 0 )
+                        {
+                            factor = SuperCellSize::toRT()[ axis ] - cell[ axis ] +
+                                thickness - 1;
+                        }
+                        else
+                        {
+                            factor = SuperCellSize::toRT()[ axis ] + cell[ axis ] -
+                                localDomainCells[ axis ] + thickness;
+                        }
+
+                        if( factor <= 0 )
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            float_X const a = math::exp( -absorberStrength * float_X( factor ) );
+                            field( cell ) = field( cell ) * a;
+                        }
+                    } while( true );
+                }
+            );
 
         }
+    };
 
-        if (factor <= float_X(0.0))
-        {
-            factor = float_X(0.0);
-            nvidia::atomicAllExch(&finish,1);
-        }
-        else
+} // namespace detail
+
+    /** damp each field component at the one exchange border
+     *
+     * @tparam T_numWorkers number of workers
+     */
+    template< uint32_t T_numWorkers >
+    struct KernelAbsorbBorder
+    {
+        /** damp a field at the border
+         *
+         * @tparam T_BoxedMemory pmacc::DataBox, type of the field
+         * @tparam T_Mapping mapper functor type
+         *
+         * @param field filed to manipulate
+         * @param thickness the thickness of the absorber area (in cells)
+         * @param absorberStrength strength of the absorber
+         * @param mapper functor to map a block to a supercell
+         */
+        template<
+            typename T_BoxedMemory,
+            typename T_Mapping
+        >
+        DINLINE void operator()(
+            T_BoxedMemory field,
```

Comment:
In my current alpaka/cupla branch I found out that non const parameter types for the kernel must be passed as copy by value. All in all I discussed this behavior with @theZiz and copy by value also for a const parameter should better perform with OpenMP than a reference because in that case the variable is private for the OpenMP thread.